server/routes.go

```diff
@@ -497,6 +497,41 @@ func (b *BaseRouter) addActivityRoutes() {
 		}
 		c.AbortWithStatusJSON(http.StatusBadRequest, ErrorResponse{Error: err.Error()})
 	})
+
+	activity.PUT(":id", func(context *gin.Context) {
+		userId := context.MustGet("userId").(uint)
+		id, err := strconv.ParseUint(context.Param("id"), 10, 32)
+		if err != nil {
+			context.Status(400)
+			return
+		}
+		var activityUpdateRequest ActivityUpdateRequest
+		err = context.ShouldBindJSON(&activityUpdateRequest)
+		if err == nil {
+			err = updateActivity(b.db, userId, uint(id), activityUpdateRequest)
+			if err != nil {
+				context.JSON(http.StatusForbidden, ErrorResponse{Error: err.Error()})
+				return
+			}
+			context.Status(http.StatusOK)
+			return
+		}
+	})
```

Comment:
After the `err == nil` check, we need to return an error similar to we do above:

```
c.AbortWithStatusJSON(http.StatusBadRequest, ErrorResponse{Error: err.Error()})
```